packages/user_packages/iotcc_mqtt_device.py

```diff
@@ -0,0 +1,190 @@
+                            'Trying Device {0} Registration failed with following error - {1}'.format(device.name,
+                                                                                                      str(e)))
+                        log.info('{0} Device Registration: Attempt: {1}'.format(device.name, str(reg_attempts)))
+                        time.sleep(delay_retries)
+
+                self.reg_devices.append(reg_device)
+                # Attempts to set device relationship with edge system
+                relationship_attempts = 0
+                while relationship_attempts <= retry_attempts:
+                    try:
+                        self.iotcc.create_relationship(self.iotcc_edge_system, reg_device)
+                        break
+                    except Exception as e:
+                        if relationship_attempts == retry_attempts:
+                            raise
+                        relationship_attempts += 1
+                        log.error(
+                            'Trying Device {0} relationship with Edge System failed with following error - {1}'.format(
+                                device.name, str(e)))
+                        log.info(
+                            '{0} Device Relationship: Attempt: {1}'.format(device.name, str(relationship_attempts)))
+                        time.sleep(delay_retries)
+                # Use the device name as identifier in the registry to easily refer the device in other packages
+                device_registry_name = socket.gethostname() + "-ChildDev" + str(i)
+                registry.register(device_registry_name, reg_device)
+
+                # Setting multiple properties by passing Dictonary object for Devices with the retry attempts
+                # in case of exceptions
+                prop_attempts = 0
+                while prop_attempts <= retry_attempts:
+                    try:
+                        self.iotcc.set_properties(reg_device,
+                                                  {"Country": "USA-G", "State": "California", "City": "Palo Alto",
+                                                   "Location": "VMware HQ", "Building": "Promontory H Lab",
+                                                   "Floor": "First Floor"})
+                        break
+                    except Exception as e:
+                        prop_attempts = prop_attempts + 1
+                        log.error('Exception while setting property for Device {0} - {1}'.format(
+                            (device.name, str(e))))
+                        log.info('Trying setting properties for Device {0}: Attempt - {1}'.format(device.name,
+                                                                                                  str(prop_attempts)))
+                        time.sleep(delay_retries)
+
+                try:
+                    # Registering Metric for Device
+                    self.metrics = []
```

Comment:
I had a discussion about this with @KohliDev. You were right, it seems self.metrics is getting re-initialized for every device and hence only the last metric will be present in self.metrics. So this also needs to be corrected.